LogStreamTest.shouldIncreasePositionOnRestart is flaky

```
Error
Expecting:
 <4294967296L>
to be greater than:
 <4294967488L>
Stacktrace
java.lang.AssertionError: 
Expecting:
 <4294967296L>
to be greater than:
 <4294967488L> 
	at io.zeebe.logstreams.log.LogStreamTest.shouldIncreasePositionOnRestart(LogStreamTest.java:111)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:377)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:138)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:465)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:451)
Standard Output
09:05:42.126 [] [main] INFO  io.zeebe.test - Test finished: shouldBuildLogStream(io.zeebe.logstreams.log.LogStreamTest)
09:05:42.126 [] [main] INFO  io.zeebe.test - Test started: shouldIncreasePositionOnRestart(io.zeebe.logstreams.log.LogStreamTest)
09:05:42.639 [Broker-0-LogStream-0] [-zb-actors-3] DEBUG io.zeebe.logstreams - Configured log appender back pressure at partition 0 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
09:05:42.733 [Broker-0-LogStream-0] [-zb-actors-4] INFO  io.zeebe.logstreams - Close appender for log stream 0
09:05:42.738 [0-write-buffer] [-zb-actors-0] DEBUG io.zeebe.dispatcher - Dispatcher closed
09:05:42.739 [Broker-0-LogStream-0] [-zb-actors-0] INFO  io.zeebe.logstreams - On closing logstream 0 close 1 readers
09:05:42.739 [Broker-0-LogStream-0] [-zb-actors-0] INFO  io.zeebe.logstreams - Close log storage with name 0
09:05:43.034 [Broker-0-LogStream-0] [-zb-actors-0] DEBUG io.zeebe.logstreams - Configured log appender back pressure at partition 0 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
09:05:43.115 [Broker-0-LogStream-0] [-zb-actors-5] INFO  io.zeebe.logstreams - Close appender for log stream 0
09:05:43.123 [0-write-buffer] [-zb-actors-0] DEBUG io.zeebe.dispatcher - Dispatcher closed
09:05:43.124 [Broker-0-LogStream-0] [-zb-actors-0] INFO  io.zeebe.logstreams - On closing logstream 0 close 1 readers
09:05:43.124 [Broker-0-LogStream-0] [-zb-actors-0] INFO  io.zeebe.logstreams - Close log storage with name 0```

--- a/logstreams/src/test/java/io/zeebe/logstreams/log/LogStreamTest.java
+++ b/logstreams/src/test/java/io/zeebe/logstreams/log/LogStreamTest.java
@@ -15,6 +15,7 @@ import static org.junit.Assert.assertNotNull;
 
 import io.zeebe.logstreams.util.LogStreamRule;
 import io.zeebe.logstreams.util.SynchronousLogStream;
+import io.zeebe.test.util.TestUtil;
 import org.agrona.DirectBuffer;
 import org.junit.Before;
 import org.junit.Rule;
@@ -100,6 +101,7 @@ public final class LogStreamTest {
     writer.value(wrapString("value")).tryWrite();
     writer.value(wrapString("value")).tryWrite();
     final long positionBeforeClose = writer.value(wrapString("value")).tryWrite();
+    TestUtil.waitUntil(() -> logStream.getCommitPosition() >= positionBeforeClose);
 
     // when
     logStream.close();

--- a/logstreams/src/test/java/io/zeebe/logstreams/util/LogStreamRule.java
+++ b/logstreams/src/test/java/io/zeebe/logstreams/util/LogStreamRule.java
@@ -61,18 +61,6 @@ public final class LogStreamRule extends ExternalResource {
     return new LogStreamRule(temporaryFolder, true, b -> {});
   }
 
-  public static LogStreamRule createRuleWithoutStarting(final TemporaryFolder temporaryFolder) {
-    return new LogStreamRule(temporaryFolder, false, b -> {});
-  }
-
-  public SynchronousLogStream startLogStreamWithStorageConfiguration(
-      final UnaryOperator<Builder> builder) {
-    logStorageRule = new AtomixLogStorageRule(temporaryFolder);
-    this.logStorageRule.open(builder);
-    createLogStream();
-    return logStream;
-  }
-
   @Override
   protected void before() {
     actorSchedulerRule = new ActorSchedulerRule(clock);

